build_tools/cmake/iree_copts.cmake

```diff
@@ -248,6 +248,8 @@ iree_select_compiler_opts(IREE_DEFAULT_COPTS
     "-Wno-unknown-pragmas"
     "-Wno-unused-but-set-variable"
     "-Wno-misleading-indentation"
+    # False positives? Not useful? https://stackoverflow.com/a/78760067
+    $<$<COMPILE_LANGUAGE:CXX>:-Wno-dangling-reference>
```

Comment:
`cc1plus: error: unrecognized command line option '-Wno-dangling-reference' [-Werror]`

argh, older versions of gcc don't recognize this option
https://github.com/iree-org/iree/actions/runs/12905050255/job/35983453401#step:4:6256

will try to fix-forward with something like `GCC_GTE_XYZ`
https://github.com/iree-org/iree/blob/a4306950c00111804967adf6b636b8146b991395/build_tools/cmake/iree_macros.cmake#L393